pandas/core/format.py

```diff
@@ -1895,6 +1895,17 @@ def get_formatted_cells(self):
 
 def format_array(values, formatter, float_format=None, na_rep='NaN',
                  digits=None, space=None, justify='right'):
+
```

Comment:
make a ``CategoricalFormatter`` (a trivial sub-class of ``GenericArray``). Then you can simply override ``_format_strings`` as needed.